pkgs/by-name/re/rerun/package.nix

```diff
@@ -18,113 +19,130 @@
   python3Packages,
+  callPackage,
 }:
-
-rustPlatform.buildRustPackage rec {
-  pname = "rerun";
-  version = "0.13.0";
-
+let
+  version = "0.17.0";
+  # they keep a rust-toolchain.toml
   src = fetchFromGitHub {
     owner = "rerun-io";
     repo = "rerun";
     rev = version;
-    hash = "sha256-HgzzuvCpzKgWC8it0PSq62hBjjqpdgYtQQ50SNbr3do=";
+    hash = "sha256-FrUxGMBau7ak9mK4MYy1LJjmTX4prMHCYYeVXJ69mdg=";
   };
-  patches = [
-    # Disables a doctest that depends on a nightly feature
-    ./0001-re_space_view_time_series-utils-patch-out-doctests-w.patch
-
-
-    # "Fix cell size test now that the overhead has shrunk"
-    # https://github.com/rerun-io/rerun/pull/5917
-    (fetchpatch {
-      url = "https://github.com/rerun-io/rerun/commit/933fc5cc1f3ee262a78bd4647257295747671152.patch";
-      hash = "sha256-jCeGfzKt0oYqIea+7bA2V/U9VIjhVvfQzLRrYG4jaHY=";
-    })
-  ];
-
-  cargoHash = "sha256-qvnkOlcjADV4b+JfFAy9yNaZGaf0ZO7hh9HBg5XmPi0=";
-
-  nativeBuildInputs = [
-    (lib.getBin binaryen) # wasm-opt
-
-    # @SomeoneSerge: Upstream suggests `mold`, but I didn't get it to work
-    lld
-
-    pkg-config
-    protobuf
-    rustfmt
-  ];
-
-  buildInputs =
-    [
-      freetype
-      glib
-      gtk3
-      (lib.getDev openssl)
-      libxkbcommon
-      vulkan-loader
-    ]
-    ++ lib.optionals stdenv.isDarwin [
-      darwin.apple_sdk.frameworks.AppKit
-      darwin.apple_sdk.frameworks.CoreFoundation
-      darwin.apple_sdk.frameworks.CoreGraphics
-      darwin.apple_sdk.frameworks.CoreServices
-      darwin.apple_sdk.frameworks.Foundation
-      darwin.apple_sdk.frameworks.IOKit
-      darwin.apple_sdk.frameworks.Metal
-      darwin.apple_sdk.frameworks.QuartzCore
-      darwin.apple_sdk.frameworks.Security
-    ]
-    ++ lib.optionals stdenv.isLinux [ (lib.getLib wayland) ];
-
-  addDlopenRunpaths = map (p: "${lib.getLib p}/lib") (
-    lib.optionals stdenv.hostPlatform.isLinux [
-      libxkbcommon
-      vulkan-loader
-      wayland
-    ]
-  );
-
-  addDlopenRunpathsPhase = ''
-    elfHasDynamicSection() {
-        patchelf --print-rpath "$1" >& /dev/null
-    }
-
-    while IFS= read -r -d $'\0' path ; do
-      elfHasDynamicSection "$path" || continue
-      for dep in $addDlopenRunpaths ; do
-        patchelf "$path" --add-rpath "$dep"
-      done
-    done < <(
-      for o in $(getAllOutputNames) ; do
-        find "''${!o}" -type f -and "(" -executable -or -iname '*.so' ")" -print0
-      done
-    )
-  '';
+  fenix = callPackage (fetchFromGitHub {
+    owner = "nix-community";
+    repo = "fenix";
+    # commit from: 2024-08-15
+    rev = "6e4233dc54850e8aff6eff401400e9a9343881eb";
+    hash = "sha256-7ehq0nfRHpU3oNAkRpklaDxQZUpuaUig2sR2LI+IL/U=";
+  }) { };
+  # use rust toolchain specified by the repo itself
+  rust = fenix.fromToolchainFile {
+    dir = "${src}";
+    sha256 = "sha256-e4mlaJehWBymYxJGgnbuCObVlqMlQSilZ8FljG9zPHY=";
+  };
+in
+(makeRustPlatform {
+  cargo = rust;
+  rustc = rust;
+}).buildRustPackage
```

Comment:
Please don't do this.  Making it work with 1.80 should not be hard.

Reply:
Actually, you _can't_ use fenix in Nixpkgs like this.  It's IFD, which is not allowed by Hydra.

Reply:
Sorry I was lazy. I'll fix it with a `time` patch 🙈 👍